Descent3/ObjInit.cpp

```diff
@@ -734,8 +734,8 @@ int ObjInitGeneric(object *objp, bool reinit) {
   // Deal with deleted type
   if (obj_info->type == OBJ_NONE) {
     int i;
-    for (i = 0, obj_info = Object_info; i < MAX_OBJECT_IDS; i++, obj_info) { // find other object of same type
-      if (Object_info[i].type == objp->type)
+    for (i = 0, obj_info = std::data(Object_info); i < MAX_OBJECT_IDS; i++, obj_info++) { // find other object of same type
```

Comment:
could this just be an iterator loop?
```suggestion
    for (auto const& obj_info : Object_info) { // find other object of same type
```

Reply:
As I stated at the beginning of the PR...
> There is more in-depth work to adapting how the arrays are used but I didn't want to make the PR too large.

I'm 100% in agreement with the suggestion but I didn't want to make a PR that required too much analysis.